README.md

```diff
@@ -220,30 +269,15 @@ includes some additional security token, its duration should be limited to this
 expiration time to prevent callers from extending their access beyond the
 allowed time.
 
-The `req.satisfies(.., noReply)` method returns `true` if the client satisfies
-one of the scopesets. If the client does not satisfy one of the scopesets, it
-returns `false` and sends an error message unless `noReply` is true. 
-
-The first argument to `req.satisfies` can be a scopeset (in disjunctive normal
-form as described above).  Or, it can be an object, in which case the method
-will assume this object is a mapping from scope parameters (`<name>`) to
-values.
-
-If `deferAuth` is set to `true`, then authentication will be postponed to the
-first invocation of `req.satisfies`.  Note that `deferAuth` will not perform
-authorization unless, `req.satisfies({})` is called either without arguments or
-with an object as first argument.  If `deferAuth` is false, then req.params
-will be used as the scope parameters.
-
-Where authorization depends on parameters or the contents of the request, the
-`req.satisfies` method is generally used with `deferAuth: true` in a construct
-like this:
+The `req.authorize(params, options)` method returns `true` if the client satisfies
```

Comment:
I propose we change this semantics...

Instead we have `req.authorize(params)`, and if authorization fails, then it throw a special error. We can then use `err instanceof AuthorizationError` to catch it in the place where `taskcluster-lib-api` calls the handler..
So if handler doesn't handle the error then it'll propagate up to tc-lib-api will which know how to catch and handle this special error class and knows how to print a nice error message...

This way we get rid of all the `if (!this.authorize(...) { return; }` statements... and it just becomes `this.authorize(...)`.
Handler implementors that wants to do something special can catch the error... and if necessary something like `try { this.authorize(...); ... } finally { ... }` will work nicely...